src/consumption.cpp

```diff
@@ -1243,9 +1243,9 @@ units::volume Character::masticated_volume( const item &food ) const
 {
     units::volume water_vol = ( food.get_comestible()->quench > 0 ) ? food.get_comestible()->quench *
                               5_ml : 0_ml;
-    units::mass food_dry_weight = units::from_gram( units::to_gram( food.weight() ) -
-                                  units::to_milliliter( water_vol ) ) / food.count();
-    units::volume food_dry_volume = ( food.volume() - water_vol ) / food.count();
+    units::mass food_dry_weight = units::from_gram( units::to_gram( food.weight() ) / food.count() -
+                                  units::to_milliliter( water_vol ) );
```

Comment:
I'm not very familiar with the codebase (nor with C++ for that matter), but this feels either wrong, or at least confusing.
My reading of this is:
1) get the weight (in unit::mass) of the single food charge
2) convert (1) to gramms
3) get the number (int?) of milliliters of water
4) convert (2) to .. dimensionless int ??..
5) subtract the two (3) from (4). I.e. subtract the number of milliliters from  number of gramms..  Which "makes sense" because water has density of 1, but ugh.. shall we maybe make that more explicit?
6) cast the integer (?) in (5) to mass (i guess assuming `123` means `123 gramms`.. ?)

May I suggest something closer to 
```suggestion
    units::mass food_dry_weight = food.weight() / food.count() -
                                  units::from_gram( 1 * units::to_milliliter( water_vol ) ) );
```
?
(or even extract the latter thing into its own `water_weight` variable for "consistency" with `water_vol`?)

Reply:
It is taking advantage of the fact that for water 1g=1ml.
https://github.com/CleverRaven/Cataclysm-DDA/blob/master/src/consumption.cpp#L1241